### ISSUE TITLE
[google_maps_flutter_web] UnimplementedError: updateTileOverlays() has not been implemented.

### ISSUE THREAD

**Issue by mrmax99** [flutter/flutter#76361]
## Steps to Reproduce

When instantiating and building GoogleMap on flutter web, it triggers a bunch of "Error: UnimplementedError: updateTileOverlays() has not been implemented." errors.

pubspec.yaml
```
dependencies:
  google_maps_flutter: ^1.2.0
  google_maps_flutter_web: ^0.1.1
```

main.dart
```
import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  Set<Marker> markers = Set<Marker>();
  

  @override
  void initState() {
    super.initState();    
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'maps web experiment',
            ),
            SizedBox(
              width:500,
              height:500,
              child: GoogleMap(
                initialCameraPosition: CameraPosition(
                  target: LatLng(0,0),
                  zoom: 10
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

```

**Expected results:** 
No error, just the map.

**Actual results:**
Map appears to work fine, but there are errors:

```
Error: UnimplementedError: updateTileOverlays() has not been implemented.
    at Object.throw_ [as throw] (http://localhost:60163/dart_sdk.js:5331:11)
    at google_maps_flutter_web.GoogleMapsPlugin.new.updateTileOverlays (http://localhost:60163/packages/google_maps_flutter_platform_interface/src/events/map_event.dart.lib.js:456:17)
    at google_maps_flutter.GoogleMapController.__.[_updateTileOverlays] (http://localhost:60163/packages/google_maps_flutter/google_maps_flutter.dart.lib.js:215:61)
    at google_maps_flutter._GoogleMapState.new._updateTileOverlays$ (http://localhost:60163/packages/google_maps_flutter/google_maps_flutter.dart.lib.js:718:40)
    at _updateTileOverlays$.next (<anonymous>)
    at http://localhost:60163/dart_sdk.js:39029:33
    at _RootZone.runUnary (http://localhost:60163/dart_sdk.js:38886:58)
    at _FutureListener.thenAwait.handleValue (http://localhost:60163/dart_sdk.js:33872:29)
    at handleValueCallback (http://localhost:60163/dart_sdk.js:34432:49)
    at Function._propagateToListeners (http://localhost:60163/dart_sdk.js:34470:17)
    at _Future.new.[_completeWithValue] (http://localhost:60163/dart_sdk.js:34312:23)
    at async._AsyncCallbackEntry.new.callback (http://localhost:60163/dart_sdk.js:34335:35)

```

<details>
  <summary>Logs</summary>

```
> flutter run -d chrome --verbose
[  +78 ms] executing: [C:\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +80 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 1fe38dcb5fc58242c6e0904a318bcc3ed5a4d837
[   +1 ms] executing: [C:\flutter/] git tag --points-at 1fe38dcb5fc58242c6e0904a318bcc3ed5a4d837
[  +70 ms] Exit code 0 from: git tag --points-at 1fe38dcb5fc58242c6e0904a318bcc3ed5a4d837
[        ] 1.26.0-17.5.pre
[  +39 ms] executing: [C:\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +53 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/beta
[        ] executing: [C:\flutter/] git ls-remote --get-url origin
[  +47 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +80 ms] executing: [C:\flutter/] git rev-parse --abbrev-ref HEAD
[  +51 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] beta
[  +70 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[  +10 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +8 ms] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +56 ms] executing: C:\Users\Max\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[  +44 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +9 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +77 ms] executing: C:\Users\Max\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[  +17 ms] Skipping pub get: version match.
[  +23 ms] Found plugin flutter_plugin_android_lifecycle at C:\flutter\.pub-cache\hosted\pub.dartlang.org\flutter_plugin_android_lifecycle-1.0.11\
[   +9 ms] Found plugin google_maps_flutter at C:\flutter\.pub-cache\hosted\pub.dartlang.org\google_maps_flutter-1.2.0\
[   +9 ms] Found plugin google_maps_flutter_web at C:\flutter\.pub-cache\hosted\pub.dartlang.org\google_maps_flutter_web-0.1.1\
[ +106 ms] Found plugin flutter_plugin_android_lifecycle at C:\flutter\.pub-cache\hosted\pub.dartlang.org\flutter_plugin_android_lifecycle-1.0.11\
[   +5 ms] Found plugin google_maps_flutter at C:\flutter\.pub-cache\hosted\pub.dartlang.org\google_maps_flutter-1.2.0\
[   +2 ms] Found plugin google_maps_flutter_web at C:\flutter\.pub-cache\hosted\pub.dartlang.org\google_maps_flutter_web-0.1.1\
[  +28 ms] Generating C:\Users\Max\Documents\bash_work\bug_web_maps_infowindow\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[  +91 ms] List of devices attached
[  +72 ms] Launching lib\main.dart on Chrome in debug mode...
[   +8 ms] List of devices attached
[ +116 ms] Updating assets
[  +85 ms] Waiting for connection from debug service on Chrome...
[  +13 ms] Found plugin flutter_plugin_android_lifecycle at C:\flutter\.pub-cache\hosted\pub.dartlang.org\flutter_plugin_android_lifecycle-1.0.11\
[   +3 ms] Found plugin google_maps_flutter at C:\flutter\.pub-cache\hosted\pub.dartlang.org\google_maps_flutter-1.2.0\
[   +3 ms] Found plugin google_maps_flutter_web at C:\flutter\.pub-cache\hosted\pub.dartlang.org\google_maps_flutter_web-0.1.1\
[  +33 ms] Found plugin flutter_plugin_android_lifecycle at C:\flutter\.pub-cache\hosted\pub.dartlang.org\flutter_plugin_android_lifecycle-1.0.11\
[   +2 ms] Found plugin google_maps_flutter at C:\flutter\.pub-cache\hosted\pub.dartlang.org\google_maps_flutter-1.2.0\
[   +2 ms] Found plugin google_maps_flutter_web at C:\flutter\.pub-cache\hosted\pub.dartlang.org\google_maps_flutter_web-0.1.1\
[ +103 ms] <- reset
[  +10 ms] C:\flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev C:\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root
C:\flutter\bin\cache\flutter_web_sdk/ --incremental --target=dartdevc --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill
C:\Users\Max\AppData\Local\Temp\flutter_tools.fbefbc4d\flutter_tool.9824093e\app.dill --libraries-spec file:///C:/flutter/bin/cache/flutter_web_sdk/libraries.json --packages
C:\Users\Max\Documents\bash_work\bug_web_maps_infowindow\.dart_tool\package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation
--filesystem-root C:\Users\Max\AppData\Local\Temp\flutter_tools.fbefbc4d\flutter_tools.8d6f19f --filesystem-scheme org-dartlang-app --initialize-from-dill
build\c7922d95bf4a2462b75c84a97c312edb.cache.dill.track.dill --platform file:///C:/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill --no-sound-null-safety
[  +16 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[+18987 ms] Waiting for connection from debug service on Chrome... (completed in 19.2s)
[   +1 ms] Synced 32.6MB.
[        ] <- accept
[   +1 ms] Caching compiled dill
[ +266 ms] Using Opening in existing browser session.

[+7006 ms] [CHROME]:
[   +1 ms] [CHROME]:DevTools listening on ws://127.0.0.1:53617/devtools/browser/6db172dc-d644-4a6b-b14b-f7d28374dc10
[+1344 ms] DwdsInjector: Received request for entrypoint at http://localhost:53461/main_module.bootstrap.js
[   +4 ms] MetadataProvider: Loading debug metadata...
[  +22 ms] MetadataProvider: Loaded debug metadata
[   +7 ms] DwdsInjector: Injected debugging metadata for entrypoint at http://localhost:53461/main_module.bootstrap.js
[+7827 ms] DevHandler: Debug service listening on ws://127.0.0.1:53668/jlFuNRHbjtg=/ws

[   +9 ms] Debug service listening on ws://127.0.0.1:53668/jlFuNRHbjtg=/ws
[   +2 ms] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[   +2 ms] Warning: Flutter's support for web development is not stable yet and hasn't
[   +1 ms] been thoroughly tested in production environments.
[  +12 ms] For more information see https://flutter.dev/web
[   +2 ms]   To hot restart changes while running, press "r" or "R".
[   +1 ms] For a more detailed help message, press "h". To quit, press "q".
[+2162 ms] Error: UnimplementedError: updateTileOverlays() has not been implemented.
                        at Object.throw_ [as throw] (http://localhost:53461/dart_sdk.js:5331:11)
                        at google_maps_flutter_web.GoogleMapsPlugin.new.updateTileOverlays
(http://localhost:53461/packages/google_maps_flutter_platform_interface/src/events/map_event.dart.lib.js:456:17)
                        at google_maps_flutter.GoogleMapController.__.[_updateTileOverlays] (http://localhost:53461/packages/google_maps_flutter/google_maps_flutter.dart.lib.js:215:61)      
                        at google_maps_flutter._GoogleMapState.new._updateTileOverlays$ (http://localhost:53461/packages/google_maps_flutter/google_maps_flutter.dart.lib.js:718:40)
                        at _updateTileOverlays$.next (<anonymous>)
                        at http://localhost:53461/dart_sdk.js:39029:33
                        at _RootZone.runUnary (http://localhost:53461/dart_sdk.js:38886:58)
                        at _FutureListener.thenAwait.handleValue (http://localhost:53461/dart_sdk.js:33872:29)
                        at handleValueCallback (http://localhost:53461/dart_sdk.js:34432:49)
                        at Function._propagateToListeners (http://localhost:53461/dart_sdk.js:34470:17)
                        at _Future.new.[_completeWithValue] (http://localhost:53461/dart_sdk.js:34312:23)
                        at async._AsyncCallbackEntry.new.callback (http://localhost:53461/dart_sdk.js:34335:35)
                        at Object._microtaskLoop (http://localhost:53461/dart_sdk.js:39173:13)
                        at _startMicrotaskLoop (http://localhost:53461/dart_sdk.js:39179:13)
                        at http://localhost:53461/dart_sdk.js:34686:9
```

```
> flutter analyze
Analyzing bug_web_maps_infowindow...                                    
No issues found! (ran in 1.6s)
```

```
> flutter doctor -v
[√] Flutter (Channel beta, 1.26.0-17.5.pre, on Microsoft Windows [Version 10.0.19042.804], locale en-US)
    • Flutter version 1.26.0-17.5.pre at C:\flutter
    • Framework revision 1fe38dcb5f (8 days ago), 2021-02-10 16:25:47 -0800
    • Engine revision d4453f6018
    • Dart version 2.12.0 (build 2.12.0-259.9.beta)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Max\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[!] IntelliJ IDEA Community Edition (version 14)
    • IntelliJ at C:\Program Files (x86)\JetBrains\IntelliJ IDEA Community Edition 14.1.3
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X This install is older than the minimum recommended version of 2017.1.0.

[√] VS Code, 64-bit edition (version 1.53.2)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.19.0

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 88.0.4324.150
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 88.0.705.68

! Doctor found issues in 1 category.
```

</details>

**Comment by pedromassangocode** [comment]
I can reproduce the issue.

```console
[+1185 ms] [CHROME]:
[   +1 ms] [CHROME]:DevTools listening on ws://127.0.0.1:54474/devtools/browser/f7dbf5b1-a5a1-46bf-86dd-11c61fa953c8
[+3230 ms] DwdsInjector: Received request for entrypoint at http://localhost:54447/main_module.bootstrap.js
[  +11 ms] MetadataProvider: Loading debug metadata...
[  +22 ms] MetadataProvider: Loaded debug metadata (no sound null safety)
[  +16 ms] DwdsInjector: Injected debugging metadata for entrypoint at http://localhost:54447/main_module.bootstrap.js
[+3325 ms] ChromeProxyService: Initializing expression compiler for main_module.bootstrap.js with sound null safety: false
[ +514 ms] DevHandler: Debug service listening on ws://127.0.0.1:54551/wPZ4aQyk8KE=/ws

[  +12 ms] Debug service listening on ws://127.0.0.1:54551/wPZ4aQyk8KE=/ws
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[   +2 ms] 🔥  To hot restart changes while running, press "r" or "R".
[        ] For a more detailed help message, press "h". To quit, press "q".
[+13147 ms] Error: UnimplementedError: updateTileOverlays() has not been implemented.
                at Object.throw_ [as throw] (http://localhost:54447/dart_sdk.js:5342:11)
                at google_maps_flutter_web.GoogleMapsPlugin.new.updateTileOverlays (http://localhost:54447/packages/google_maps_flutter_platform_interface/src/events/map_event.dart.lib.js:456:17)
                at google_maps_flutter.GoogleMapController.__.[_updateTileOverlays] (http://localhost:54447/packages/google_maps_flutter/google_maps_flutter.dart.lib.js:215:61)
                at google_maps_flutter._GoogleMapState.new._updateTileOverlays$ (http://localhost:54447/packages/google_maps_flutter/google_maps_flutter.dart.lib.js:718:40)
                at _updateTileOverlays$.next (<anonymous>)
                at http://localhost:54447/dart_sdk.js:39157:33
                at _RootZone.runUnary (http://localhost:54447/dart_sdk.js:39014:58)
                at _FutureListener.thenAwait.handleValue (http://localhost:54447/dart_sdk.js:34000:29)
                at handleValueCallback (http://localhost:54447/dart_sdk.js:34560:49)
                at Function._propagateToListeners (http://localhost:54447/dart_sdk.js:34598:17)
                at _Future.new.[_completeWithValue] (http://localhost:54447/dart_sdk.js:34440:23)
                at async._AsyncCallbackEntry.new.callback (http://localhost:54447/dart_sdk.js:34463:35)
                at Object._microtaskLoop (http://localhost:54447/dart_sdk.js:39301:13)
                at _startMicrotaskLoop (http://localhost:54447/dart_sdk.js:39307:13)
                at http://localhost:54447/dart_sdk.js:34814:9
```

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel master, 1.27.0-5.0.pre.45, on macOS 11.2.1 20D74 darwin-x64, locale en-AO)
    • Flutter version 1.27.0-5.0.pre.45 at /Users/pedromassango/Code/flutter_master
    • Framework revision 90531179f5 (15 minutes ago), 2021-02-17 23:21:02 -0800
    • Engine revision 6993cb229b
    • Dart version 2.13.0 (build 2.13.0-30.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 53.1.5
    • Dart plugin version 203.6912

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • sdk gphone x86 64 (mobile) • emulator-5554 • android-x64    • Android 11 (API 30) (emulator)
    • macOS (desktop)            • macos         • darwin-x64     • macOS 11.2.1 20D74 darwin-x64
    • Chrome (web)               • chrome        • web-javascript • Google Chrome 88.0.4324.182

! Doctor found issues in 1 category.
```

</details>

**Comment by xOldeVx** [comment]
any solution?

**Comment by joaquinperaza** [comment]
Having the same problems

**Comment by joaquinperaza** [comment]
I think we need this published (1.2.0)
[https://github.com/flutter/plugins/commit/6fa2ead23e8d58de047f38c0f90ce5e218ef8ea5
](url)

**Comment by zackygaurav** [comment]
Any idea how to resolve this? I am stuck with the same error.

**Comment by joaquinperaza** [comment]
> Any idea how to resolve this? I am stuck with the same error.

I downgraded google_maps to 1.1.0

**Comment by huemulDeveloper** [comment]
same Error: UnimplementedError: updateTileOverlays() has not been implemented.

**Comment by stuartmorgan** [comment]
@ditman This was added recently (https://github.com/flutter/flutter/issues/25942), but hasn't been implemented for web. It's called on each update, so if it's not going to be added in the short term perhaps a temporary no-op implementation in web would be a good idea to avoid log spam.

**Comment by mrmax99** [comment]
It's an error, so a bit worse than log spam

**Comment by stuartmorgan** [comment]
The error in this case is just that the feature isn't implemented; nothing should be broken by the fact that it logs this error.

**Comment by ditman** [comment]
The default behavior of unimplemented methods is an exception. I'll get to google maps web very soon, will noop this new method :)

**Comment by ShahinTaj01** [comment]
I can pick this issue

**Comment by ditman** [comment]
@ShahinTaj01 thanks! Hit it! Remember to add me or cc me in your PR when you need review!

**Comment by ShahinTaj01** [comment]
@ditman, I just verified there are two other methods which are also unimplemented, takeSnapshot() and clearTileCache()
Do you think we should include it in the same PR ?

**Comment by ditman** [comment]
@ShahinTaj01 I think clearTileCache should be implemented, since it has to do with the "Tiles" functionality in general. We've never received any complaints about `takeSnapshot` crashing, so I'd leave it unimplemented.

**Comment by ditman** [comment]
The fix for this has been published in google_maps_flutter_web 0.2.0!

https://pub.dev/packages/google_maps_flutter_web/versions/0.2.0

(This needs to be used in conjunction with `google_maps_flutter: ^2.0.0`)

Resolving. Feel free to reopen if this doesn't solve the issue!
